_drafts/release-workflows.md

```diff
@@ -0,0 +1,383 @@
+| **push** to `next`                                    | Release on `next` channel                                                                                                                                                                                                                                                                                 |
+| **merge** any branch to `next`                        | Release on `next` channel                                                                                                                                                                                                                                                                                 |
+| **merge** some commits from `next` branch to `master` | Get the last release for commits on branch `master` (including the one that come from `next`), analyze commits, increase the version; If the commit associated with the last release found is also present on `next`, make the version available on the `latest` channel, otherwise do a regular release. |
+
+##### Future branches configuration
+
+```json
+"release": {
+  "branches": ["master", {
+		"branch": "next",
+		"type": "major"
+		}]
+}
+```
+
+The `type` is required for future branches and indicate that its reserved for a certain type of release (`minor` or `major`).
+That defines how much ahead the future branch must be versus the pervious branch in the list. If configured with `major`, the `next` branch must always one `major` release ahead of `master`.
+
+Example 1:
+- if the last release on `latest` is `2.1.0` and `next` is configured with `"type" : "major"`
+- and the last release on `next` is also `2.1.0`
+- then any releases are allowed on `latest`
+- And only `major` releases are allowed on `next`
+
+Example 2:
+- if the last release on `latest` is `2.1.0` and `next` is configured with `"type" : "major"`
+- and the last release on `next` is `2.5.0`
+- then `patch` releases are allowed on `latest`, but `minor` and `major` are forbidden
+- And any releases are allowed on `next`
+
+Example 3:
+- if the last release on `latest` is `2.1.0` and `next` is configured with `"type" : "major"`
+- and the last release on `next` is `3.0.0`
+- then `patch` and `minor` releases are allowed on `latest`, but `major` are forbidden
+- And any releases are allowed on `next`
+
+Example 4:
+- if the last release on `latest` is `3.0.0` and `next` is configured with `"type" : "major"`
+- and the last release on `next` is `2.5.0`
+- And any releases are allowed on `latest`
+- No releases are allowed on `next` (it's considered stalled) until `master` is merged into `next`. Once it's done the situation becomes the one described in Example 1
```

Comment:
since there have already been publishes from that branch, wouldnt you want to avoid rebasing? then the branch would have commits that weren't there at the time of publish and therefore didnt make it into the published version

Reply:
The idea is that rebase or merging doesn't matter in our case.
What matter is that both branches are equals (which we check by comparing the tags on both branches).
So if you have the release `2.0.0` done `next` you have a corresponding tag. If the tag is on `master` as well then the release exist in `master`, no matter if the commit referencing the tag `v2.0.0` has been merged or rebased.